### ISSUE TITLE
useRequest: staleTime should be ignored when error happens

### ISSUE THREAD

**Issue by KrishnaPG** [alibaba/hooks#950]
When `staleTime` is used and error happens in the `useRequest`, then the error is cached and returned on the reRender - which is not much useful.  It is preferable not to cache the errors and retry the requests on component update.

Is there anyway to ignore the `staleTime` when error happens and force a retry for the request? 

````
    const { data, error, loading } = useRequest('/uploads/csvHeaders/`,
      {
        cacheKey: `csvHeaders`,
        staleTime: -1, // cache never expires. If error happens, this stores the error, which is BAD
        throwOnError: false,
      },
````

In other words, `staleTime` should be applied only on successful data retrieval.

**Comment by KrishnaPG** [comment]
This is especially important when disk-based caching supported in future.

**Comment by brickspert** [comment]
Thanks for you feedback, we will redesign this API on next version which we are working.

**Comment by brickspert** [comment]
Hello：

这个问题我们已经在 v3.0 版本彻底修复，目前 v3.0 版本已经发布 alpha 版本。

欢迎试用。

文档：https://ahooks-next.surge.sh/
源码：https://github.com/alibaba/hooks/tree/next

安装最新分支
```
$ npm install --save ahooks@next
# or
$ yarn add ahooks@next
```

We have fixed this problem in the v3.0 version, and the alpha has been released. 

Welcome to try.

Doc：https://ahooks-next.surge.sh/
Source Code：https://github.com/alibaba/hooks/tree/next

Install v3.0

```
$ npm install --save ahooks@next
# or
$ yarn add ahooks@next
```
